src/control/conf.c

```diff
@@ -310,6 +310,23 @@ int dt_conf_get_bool(const char *name)
   return val;
 }
 
+gchar *dt_conf_get_path(const char *name)
+{
+  char *path = g_strdup(_conf_get_var(name));
+
+  if(!g_file_test(path, G_FILE_TEST_IS_DIR))
+  {
+    const dt_confgen_value_t *item = g_hash_table_lookup(darktable.conf->x_confgen, name);
+    if(item->type == DT_PATH)
```

Comment:
This is the part to identify whats wrong :-)

Adding `printf(stderr, "checking path item=%d of %d\n", item->type, DT_PATH);` before the test outputs `checking path item=5 of 4` which is wrong.  The 'x_confgen' returns the wrong type, i just commented the `if(item->type == DT_PATH)` line and everything works as expected.

BTW: to trigger the issue manually have some random string like
`ui_last/import_last_directory=fdahbphbbtbbrnzn`

Reply:
> Adding `printf(stderr, "checking path item=%d of %d\n", item->type, DT_PATH);` before the test outputs `checking path item=5 of 4` which is wrong.

Because the value should be `dir` and not `path` in `darktableconfig.xml.in` and there is no way in a DTD to restrict values in node. Will fix.

Reply:
Fixed.